packages/babel-plugin-transform-function-sent/src/index.js

```diff
@@ -1,59 +1,55 @@
 import syntaxFunctionSent from "babel-plugin-syntax-function-sent";
 import wrapFunction from "babel-helper-wrap-function";
 
-const yieldVisitor = {
-  Function(path) {
-    path.skip();
-  },
-  YieldExpression(path) {
-    path.replaceWith(
-      this.types.assignmentExpression("=", this.sentId, path.node),
-    );
-    path.skip();
-  },
-};
-
 export default function({ types: t }) {
+  const isFunctionSent = node =>
+    t.isIdentifier(node.meta, { name: "function" }) &&
+    t.isIdentifier(node.property, { name: "sent" });
+
+  const yieldVisitor = {
+    Function(path) {
+      path.skip();
+    },
+
+    YieldExpression(path) {
+      const replaced = t.isAssignmentExpression(path.parent, {
```

Comment:
We can't: if yield's argument contains `function.sent`, it needs to be traversed.